src/SprykerShop/Yves/LanguageSwitcherWidget/Widget/LanguageSwitcherWidget.php

```diff
@@ -146,7 +146,7 @@ protected function attachLocaleUrlsToLanguages(array $locales, string $requestUr
      */
     protected function replaceCurrentUrlLanguage(string $currentUrl, array $languages, string $replacementLanguage): string
     {
-        if (preg_match('~/(' . implode('|', $languages) . ')/~', $currentUrl)) {
+        if (preg_match('~^/(' . implode('|', $languages) . ')/?~', $currentUrl)) {
```

Comment:
@victor-ufg Could you please add test for the updated regular expression?
Ideally it should be two tests: for old non-working(incorrectly worked) one and for new one. Say positive and negative tests.